rust/nft-candy-machine/src/lib.rs

```diff
@@ -273,7 +279,11 @@ pub mod nft_candy_machine {
         for i in 0..4 {
             data[vec_start + i] = as_bytes[i]
         }
-
+        // previous CLI versions used a smaller configuration account, that's why we must check
+        // the configuration account size before writing this information
+        if data.len() >= vec_start + 5 {
```

Comment:
So I dislike this design pattern because it violates the borsh specification. The reason the number of lines u32 is right before the actual array is because that's how borsh stores vecs. Plopping this in between, while convenient, breaks things. What I would do instead is check if the date len is longer before you set the max number of lines, and if it is, offset the max number of lines by 1 byte and set it, so you can place the new boolean in that extra byte area.

Reply:
Sure, I'll change that asap, thanks for the feedback!

Reply:
Just pushed a commit that changes the position of this boolean, it should be OK even if I have to admit I'm not sure to have fully understanded the role of this mask.

Also rebased the branch on `master`. Thanks!